### PR TITLE
Add environment cleanup command

### PR DESCRIPTION
**Pull request by fzhadaev**
Added command and script for controller.

### CODE REVIEW

File: octane/util/env.py [hunk 1]
```diff
@@ -28,18 +28,22 @@
...
             continue
-        if ('controller' in node.data['roles'] or
-                'controller' in node.data['pending_roles']):
-            yield node
-            found = True
-    if not found:
+        for role in roles:
+            if (role in node.data['roles'] or
+                    role in node.data['pending_roles']):
+                yield node
```
Comment (YorikSar) [inline octane/util/env.py]: Add `break` after `yield` to not return one node twice.

File: octane/commands/cleanup.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        script_dst_filename,
+    )
+    data_file = sftp.open(data_filename, 'w')
+    data_file.write(jsonutils.dumps(cleaning_data))
+    data_file.close()
+    print(ssh.call_output(['python', script_dst_filename,
+                           data_filename], node=controller))
+
+    print(ssh.call_output(['rm', '-f', script_dst_filename], node=controller))
+    print(ssh.call_output(['rm', '-f', data_filename], node=controller))
```
Comment (YorikSar) [inline octane/commands/cleanup.py]: Use `ssh.call` if you don't need output. It'll log all output.

File: octane/util/env.py [hunk 3]
```diff
@@ -28,18 +28,22 @@
...
+
+
+def get_controllers(env):
+    controllers = get_nodes(env, ['controller'])
+    if not controllers:
```
Comment (YorikSar) [inline octane/util/env.py]: This condition will never be met since `controllers` is a generator.

File: octane/commands/cleanup.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    script_dst_filename = '/tmp/{0}'.format(script_filename)
+    data_filename = '/tmp/clean_conf.json'
+
+    sftp.put(
+        os.path.join(magic_consts.CWD, "helpers/{0}".format(script_filename)),
+        script_dst_filename,
+    )
+    data_file = sftp.open(data_filename, 'w')
+    data_file.write(jsonutils.dumps(cleaning_data))
+    data_file.close()
```
Comment (YorikSar) [inline octane/commands/cleanup.py]: Use `with` to work with files and close them:

    with sftp.open(...) as f:
        f.write(...)

File: octane/commands/cleanup.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        astute['network_metadata']['vips']['management']['ipaddr'])
+
+
+def cleanup_environment(env_id):
+    cleaning_data = {}
+    env = objects.Environment(env_id)
+    astute = env_util.get_astute_yaml(env)
+
+    access_data = get_access_data(astute)
+    access_data['auth_url'] = get_auth_url(astute)
```
Comment (YorikSar) [inline octane/commands/cleanup.py]: Can't we get it from `openrc`?

Comment (fzhadaev) [inline octane/commands/cleanup.py]: IMO it's easier to get all access params in one place.

Comment (YorikSar) [inline octane/commands/cleanup.py]: We have a common pattern of running scripts like `. openrc; python /tmp/anotherscript.py`. It's way simpler on both sides, so let's follow it here if it works.

File: octane/commands/cleanup.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        script_dst_filename,
+    )
+
+    command = ['sh', '-c', '. /root/openrc; export OS_PASSWORD=admin; python '
+               + script_dst_filename]
+
+    with ssh.popen(command, node=controller, stdin=ssh.PIPE) as proc:
+        roles = ["controller", "compute"]
+        for node in env_util.get_nodes(env, roles):
+            proc.stdin.write(node.data['fqdn']+"\n")
```
Comment (ogelbukh) [inline octane/commands/cleanup.py]: With 'fqdn' you'll get the full domain name: 'node-2.test.domain.local'. You should use node.data['hostname'] to get hostname that will appear in Nova service object: 'node-2'.

**Comment by ogelbukh** [comment]
Other than using improper param to get the host name, looks good to me. @fzhadaev plz fix that, or tell me if I'm missing something. @YorikSar could you please take a final look, and I will merge, if you're OK with it.

**Pull request merged**